lib/src/codec/udp/message_encoder.dart

```diff
@@ -106,6 +117,61 @@ class UdpMessageEncoder implements MessageEncoder {
     return writer.toByteArray();
   }
 
+  /// Determine the token length.
+  ///
+  /// The token length can either be of zero to eight bytes or be extended,
+  /// following [RFC 8974].
+  ///
+  /// [RFC 8974]: https://datatracker.ietf.org/doc/html/rfc8974
+  static int _getTokenLength(final Uint8Buffer? token) {
+    final tokenLength = token?.length ?? 0;
+    if (tokenLength <= 12) {
+      return tokenLength;
+    } else if (tokenLength <= 255 + 13) {
+      return 13;
+    } else if (tokenLength <= 65535 + 269) {
```

Comment:
Not sure we can set the limit this high? Keep in mind that 65 535 is the MTU for an IP packet, the token length limit at 65535+269 would be larger than that.

This limit also includes IP headers which are added at the network layer. The _payload_ limit, which concerns us, I think is even lower:
> A UDP datagram is carried in a single IP packet and is hence limited to a maximum payload of 65,507 bytes for IPv4 and 65,527 bytes for IPv6. The transmission of large IP packets usually requires IP fragmentation.

https://erg.abdn.ac.uk/users/gorry/course/inet-pages/udp.html

Reply:
Hmm, this should be the max value from [RFC 8974](https://datatracker.ietf.org/doc/html/rfc8974#section-2.1). However, there they state the following:

> The new definition of the "TKL" field increases the maximum token
> length that can be represented in a message to 65804 bytes.  However,
> the maximum token length that sender and recipient implementations
> support may be shorter.  For example, a constrained node of Class 1
> [[RFC7228](https://datatracker.ietf.org/doc/html/rfc7228)] might support extended token lengths only up to 32 bytes.
> 
> In CoAP over UDP, it is often beneficial to keep CoAP messages small
> enough to avoid IP fragmentation.  The maximum practical token length
> may therefore also be influenced by the Path MTU (PMTU).  See
> [Section 4.6 of [RFC7252]](https://datatracker.ietf.org/doc/html/rfc7252#section-4.6) for details.

Should we enforce a hard limit here? Or should we let users decide to handle this?

Reply:
Huh, odd, let's go with the RFC then, I'm sure they have their reasons